Core/src/Vertexing/AdaptiveGridTrackDensity.cpp

```diff
@@ -12,16 +12,39 @@
 
 #include <algorithm>
 
-float Acts::AdaptiveGridTrackDensity::getBinCenter(int bin, float binExtent) {
+namespace Acts {
+
+double AdaptiveGridTrackDensity::getBinCenter(int bin, double binExtent) {
   return bin * binExtent;
 }
 
-int Acts::AdaptiveGridTrackDensity::getBin(float value, float binExtent) {
+int AdaptiveGridTrackDensity::getBin(double value, double binExtent) {
   return static_cast<int>(std::floor(value / binExtent - 0.5) + 1);
 }
 
-typename Acts::AdaptiveGridTrackDensity::DensityMap::const_iterator
-Acts::AdaptiveGridTrackDensity::highestDensityEntry(
+int AdaptiveGridTrackDensity::getTrkGridSize(double sigma, double trkSigmas,
+                                             double binExtent) {
+  int size = static_cast<int>(std::ceil(2 * trkSigmas * sigma / binExtent));
+  if (size % 2 == 0) {
+    ++size;
+  }
+  return size;
+}
+
+int AdaptiveGridTrackDensity::getSpatialTrkGridSize(double sigma) const {
+  return getTrkGridSize(sigma, m_cfg.spatialTrkSigmas, m_cfg.spatialBinExtent);
+}
+
+int AdaptiveGridTrackDensity::getTemporalTrkGridSize(double sigma) const {
```

Comment:
This function should return 0 if `sigma == 0` (i.e., if we only do spatial seeding)